### PR TITLE
Allow dynamic export of database objects

### LINKED ISSUE #2516: META: Large synthetic dataset for performance evaluation
This issue is about **research** into improving the development workflow when interested in performance bottlenecks. While we could just create a copy of the live system for local experimentation (the same as we do with the test system every so often), it might contain personal information which we rather don't want to even be able to obtain as developers.

- **Devise a method of generating reasonably authentic data** in comparison with the live system, namely a similar number of regions with similar content.
- **Verify that** a dev environment with that data **behaves similarly** to the live environment in performance (mind resources available, which might vary greatly between the machines used for development; while e.g. the redis cache is probably only to be mentioned to the developer as a potentially deciding if not active).

This should be separate from the existing `test_data.json` fixture, as the small dataset is highly preferable during quick iterations on a feature, except when performance with large data is the focus. The developer should be able to switch between them with relative ease.

### PR DESCRIPTION
**Pull request by david-venhoff**
### Short description

This is an experimental pr which adds a new api endopint that creates a data export which can be imported using `./tools/integreat-cms-cli loaddata path/to/json`

To preserve user privacy, only whitelisted models get exported and user account information will not be included in the export. Also only administrators are allowed to use this endpoint. 

This is currently a proof of concept and only exports regions, languages, pages and page translations

Please take a look at this and let me know whether this approach is something that could be incorporated :)

### Proposed changes

- Add a `dump` function that exports all objects of selected models as json.
- Add a new api endpoint `/v3/database` which uses this function to create a data export.

### Resolved issues

Fixes: #2516

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

File: integreat_cms/locale/de/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -9582,6 +9582,10 @@ msgstr "Englisch"
 msgid "Dutch"
 msgstr "Niederländisch"
 
+#: data_dump/apps.py
+msgid "Data dump"
```
Comment (ulliholtgrave) [inline integreat_cms/locale/de/LC_MESSAGES/django.po:9586]: I guess I prefer the term "Database dump" and "Datenbank-Export" in German.

**Review by ulliholtgrave — CHANGES_REQUESTED** [review]
Nice idea and so far it works as expected. However, I have 2 questions:
1. I did test it with our default test data and it was sufficiently fast, but did you do some performance bench marking for a  bigger database?
2. We generally want to use this file as an import for our local setups, right?

**Comment by david-venhoff** [comment]
> I did test it with our default test data and it was sufficiently fast, but did you do some performance bench marking for a bigger database?

No, I did not test it with a bigger database. I do expect it to be quite slow however, since we have so many page translations on the production database and a for-loop that has to go over every single one of them. Not sure if there is a way to make it faster.
I would argue that it should be fine however, since this operation should usually only needed once every few months per developer probably.

> We generally want to use this file as an import for our local setups, right?

Probably only for performance testing, unless we can optimize the cms enough that working with realistic test data is bearable.

**Review by ulliholtgrave — APPROVED** [review]
Thank! I was able to export a dataset and import it :+1: 
I guess we still have to think about how and in which form we want to make them available to our devs, but this is a good start

**Comment by david-venhoff** [comment]
We decided to close this, because it is also possible to use database dumps, and the advantage of this approach is not really big enough to justify it

**Pull request closed without merging**